flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -98,14 +100,11 @@
 
 	private final BufferManager bufferManager;
 
-	/**
-	 * Indicates the last overtaken sequence number by the most recent {@link CheckpointBarrier}
-	 * before task thread started checkpoint, or {@code null} if {@link CheckpointBarrier} hasn't
-	 * arrived yet.
-	 */
 	@GuardedBy("receivedBuffers")
-	@Nullable
-	private Integer lastOvertakenSequenceNumber = null;
+	private int lastBarrierSequenceNumber = NONE;
```

Comment:
Ah that what I was missing in the first main commit.
I propose to squash the 2 main commits or at least already use the final name/type in the first main commit, so that the two commits don't touch every piece of code twice.

Reply:
The name of this field is also changed, so migrating the type change to the older commit will only eliminate two changes in the 2nd commit. 
Besides that, I'd like to keep the first commit (from the other PR) separate and untouched to avoid confusion.

Reply:
As a reviewer of this PR, I'm more confused by separation. I think Piotr is also not a big fan of touching the same code in multiple commits. The question is which confusion you want to avoid? Is it about code ownership? Then, I wouldn't mind keeping it separate.

Reply:
It would be easier I think for us to unroll the history later if needed.
But I don't insist, moved type change to Piotr's commit.